lib/Parse/ParseExpr.cpp

```diff
@@ -1418,35 +1442,23 @@ ParserResult<Expr> Parser::parseExprPostfix(Diag<> ID, bool isExprBasic) {
         continue;
       }
 
-      // Handle "x.dynamicType" - A metatype expr.
-      // Deprecated in SE-0096: `x.dynamicType` becomes `type(of: x)`
-      //
-      // FIXME(SE-0096): This will go away along with the keyword soon.
-      if (Tok.is(tok::kw_dynamicType)) {
-        // Fix-it
+      // Handle "x.self" expr.
+      if (Tok.is(tok::kw_self)) {
+        Result = makeParserResult(
+          new (Context) DotSelfExpr(Result.get(), TokLoc, consumeToken()));
+        continue;
+      }
+
+      // Handle the deprecated 'x.dynamicType' and migrate it to `type(of: x)`
+      if (Tok.getText() == "dynamicType") {
         auto range = Result.get()->getSourceRange();
         auto dynamicTypeExprRange = SourceRange(TokLoc, consumeToken());
         diagnose(TokLoc, diag::expr_dynamictype_deprecated)
           .highlight(dynamicTypeExprRange)
           .fixItReplace(dynamicTypeExprRange, ")")
           .fixItInsert(range.Start, "type(of: ");
-
-        // HACK: Arbitrary.
-        auto loc = range.Start;
-        auto dt = new (Context) DynamicTypeExpr(loc, loc, Result.get(), loc, Type());
-        dt->setImplicit();
-        Result = makeParserResult(dt);
```

Comment:
If it causes you too much pain it can probably be removed.  Much as I think it's nice to have this migration aid, the user will receive a "no such member" error even if it isn't here which should be hint enough that `dynamicType` is gone.  A special case can be added to Sema to diagnose this too.